### PR TITLE
Upgrade to Rails 5.1 and deprecate old resources

### PR DESCRIPTION
**Pull request by ninjapanzer**
### What am I?

I hit a lot of issues when working in my docker branch so I decided to take a step back and clean things up at the core and this led to upgrading to rails 5.1 which was not too tough but now I will be revising the docker upgrade stuff since some of the yarn issues are built into rails now. Also I will be following a different set of instructions from the upgrade passenger gem

### CODE REVIEW

File: bin/update [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+# path to your application root.
+APP_ROOT = Pathname.new File.expand_path('../../', __FILE__)
+
+def system!(*args)
+  system(*args) || abort("\n== Command #{args} failed ==")
+end
+
+chdir APP_ROOT do
+  # This script is a way to update your development environment automatically.
```
Comment (ninjapanzer) [inline bin/update]: Similar code found in setup:14, update:14 (mass = 50)

File: Gemfile [hunk 2]
```diff
@@ -30,12 +28,11 @@ gem 'bourbon', '~> 4.2.3'
...
 
-gem 'haml', '~> 4.0.6'
+gem 'haml', '~> 5.0.1'
 
-gem 'refile', '~> 0.5', require: "refile/rails"
-gem 'refile-mini_magick', git: "https://github.com/ninjapanzer/refile-mini_magick.git", branch: "pks/add_resample_as_a_transform"
-
-gem 'browserify-rails', '~> 1.2.0'
+gem 'refile', github: 'refile/refile', require: 'refile/rails'
+gem 'image_processing'
```
Comment (ninjapanzer) [inline Gemfile]: Gems should be sorted in an alphabetical order within their section of the Gemfile. Gem `image_processing` should appear before `refile`.

File: app/controllers/reviews_controller.rb [hunk 3]
```diff
@@ -1,6 +1,6 @@
 class ReviewsController < ApplicationController
 
-  before_filter :authenticate!, except:[:show, :index]
+  before_action :authenticate!, except:[:show, :index]
```
Comment (ninjapanzer) [inline app/controllers/reviews_controller.rb]: Space missing after colon.

Comment (ninjapanzer) [inline app/controllers/reviews_controller.rb]: Use `%i` or `%I` for an array of symbols.

File: app/controllers/sessions_controller.rb [hunk 4]
```diff
@@ -1,17 +1,14 @@
 class SessionsController < ApplicationController
   def create
-    user = User.from_omniauth(env["omniauth.auth"])
+    user = User.from_omniauth(request.env["omniauth.auth"])
```
Comment (ninjapanzer) [inline app/controllers/sessions_controller.rb]: Prefer single-quoted strings when you don't need string interpolation or special symbols.

File: app/models/application_record.rb [hunk 5]
```diff
@@ -0,0 +1,3 @@
+class ApplicationRecord < ActiveRecord::Base
```
Comment (ninjapanzer) [inline app/models/application_record.rb]: Missing top-level class documentation comment.

Comment (ninjapanzer) [inline app/models/application_record.rb]: Missing top-level class documentation comment.

Comment (ninjapanzer) [inline app/models/application_record.rb]: Missing magic comment `# frozen_string_literal: true`.

Comment (ninjapanzer) [inline app/models/application_record.rb]: Missing top-level class documentation comment.

Comment (ninjapanzer) [inline app/models/application_record.rb]: Missing magic comment `# frozen_string_literal: true`.

File: app/models/portfolio_image.rb [hunk 6]
```diff
@@ -1,4 +1,4 @@
-class PortfolioImage < ActiveRecord::Base
+class PortfolioImage < ApplicationRecord
```
Comment (ninjapanzer) [inline app/models/portfolio_image.rb]: Missing top-level class documentation comment.

File: app/models/application_record.rb (hunk 5 above)
Comment (ninjapanzer) [inline app/models/application_record.rb]: Missing magic comment `# frozen_string_literal: true`.

File: app/models/portfolio_image.rb (hunk 6 above)
Comment (ninjapanzer) [inline app/models/portfolio_image.rb]: Missing top-level class documentation comment.

Comment (ninjapanzer) [inline app/models/portfolio_image.rb]: Missing magic comment `# frozen_string_literal: true`.

Comment (ninjapanzer) [inline app/models/portfolio_image.rb]: Missing top-level class documentation comment.

Comment (ninjapanzer) [inline app/models/portfolio_image.rb]: Missing magic comment `# frozen_string_literal: true`.

File: app/models/post.rb [hunk 7]
```diff
@@ -1,4 +1,4 @@
-class Post < ActiveRecord::Base
+class Post < ApplicationRecord
```
Comment (ninjapanzer) [inline app/models/post.rb]: Missing top-level class documentation comment.

File: app/models/portfolio_image.rb (hunk 6 above)
Comment (ninjapanzer) [inline app/models/portfolio_image.rb]: Missing magic comment `# frozen_string_literal: true`.

File: app/models/post.rb (hunk 7 above)
Comment (ninjapanzer) [inline app/models/post.rb]: Missing top-level class documentation comment.

Comment (ninjapanzer) [inline app/models/post.rb]: Missing magic comment `# frozen_string_literal: true`.

Comment (ninjapanzer) [inline app/models/post.rb]: Missing top-level class documentation comment.

Comment (ninjapanzer) [inline app/models/post.rb]: Missing magic comment `# frozen_string_literal: true`.

File: app/models/post_attachment.rb [hunk 8]
```diff
@@ -1,4 +1,4 @@
-class PostAttachment < ActiveRecord::Base
+class PostAttachment < ApplicationRecord
```
Comment (ninjapanzer) [inline app/models/post_attachment.rb]: Missing top-level class documentation comment.

File: app/models/post.rb (hunk 7 above)
Comment (ninjapanzer) [inline app/models/post.rb]: Missing magic comment `# frozen_string_literal: true`.

File: app/models/post_attachment.rb (hunk 8 above)
Comment (ninjapanzer) [inline app/models/post_attachment.rb]: Missing top-level class documentation comment.

Comment (ninjapanzer) [inline app/models/post_attachment.rb]: Missing magic comment `# frozen_string_literal: true`.

Comment (ninjapanzer) [inline app/models/post_attachment.rb]: Missing magic comment `# frozen_string_literal: true`.

Comment (ninjapanzer) [inline app/models/post_attachment.rb]: Missing top-level class documentation comment.

File: app/models/review.rb [hunk 9]
```diff
@@ -1,2 +1,2 @@
-class Review < ActiveRecord::Base
+class Review < ApplicationRecord
```
Comment (ninjapanzer) [inline app/models/review.rb]: Missing magic comment `# frozen_string_literal: true`.

Comment (ninjapanzer) [inline app/models/review.rb]: Missing magic comment `# frozen_string_literal: true`.

File: app/models/post_attachment.rb (hunk 8 above)
Comment (ninjapanzer) [inline app/models/post_attachment.rb]: Missing magic comment `# frozen_string_literal: true`.

File: app/models/seo_link.rb [hunk 10]
```diff
@@ -1,4 +1,4 @@
-class SEOLink < ActiveRecord::Base
+class SEOLink < ApplicationRecord
```
Comment (ninjapanzer) [inline app/models/seo_link.rb]: Missing top-level class documentation comment.

Comment (ninjapanzer) [inline app/models/seo_link.rb]: Missing top-level class documentation comment.

File: app/models/review.rb (hunk 9 above)
Comment (ninjapanzer) [inline app/models/review.rb]: Missing magic comment `# frozen_string_literal: true`.

File: app/models/seo_link.rb (hunk 10 above)
Comment (ninjapanzer) [inline app/models/seo_link.rb]: Missing magic comment `# frozen_string_literal: true`.

Comment (ninjapanzer) [inline app/models/seo_link.rb]: Missing magic comment `# frozen_string_literal: true`.

File: app/models/user.rb [hunk 11]
```diff
@@ -1,4 +1,4 @@
-class User < ActiveRecord::Base
+class User < ApplicationRecord
```
Comment (ninjapanzer) [inline app/models/user.rb]: Missing top-level class documentation comment.

Comment (ninjapanzer) [inline app/models/user.rb]: Missing magic comment `# frozen_string_literal: true`.

File: bin/update [hunk 12]
```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env ruby
```
Comment (ninjapanzer) [inline bin/update]: Missing magic comment `# frozen_string_literal: true`.

File: bin/yarn [hunk 13]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env ruby
```
Comment (ninjapanzer) [inline bin/yarn]: Missing magic comment `# frozen_string_literal: true`.

File: bin/yarn [hunk 14]
```diff
@@ -0,0 +1,11 @@
...
+VENDOR_PATH = File.expand_path('..', __dir__)
+Dir.chdir(VENDOR_PATH) do
+  begin
+    exec "yarnpkg #{ARGV.join(" ")}"
```
Comment (ninjapanzer) [inline bin/yarn]: Prefer single-quoted strings inside interpolations.

File: bin/yarn [hunk 15]
```diff
@@ -0,0 +1,11 @@
...
+  rescue Errno::ENOENT
+    $stderr.puts "Yarn executable was not detected in the system."
```
Comment (ninjapanzer) [inline bin/yarn]: Prefer single-quoted strings when you don't need string interpolation or special symbols.

File: bin/yarn [hunk 16]
```diff
@@ -0,0 +1,11 @@
...
+    $stderr.puts "Download Yarn at https://yarnpkg.com/en/docs/install"
```
Comment (ninjapanzer) [inline bin/yarn]: Prefer single-quoted strings when you don't need string interpolation or special symbols.

File: config/application.rb [hunk 17]
```diff
@@ -1,4 +1,4 @@
-require File.expand_path('../boot', __FILE__)
+require_relative 'boot'
```
Comment (ninjapanzer) [inline config/application.rb]: Missing magic comment `# frozen_string_literal: true`.

File: config/boot.rb [hunk 18]
```diff
@@ -1,3 +1,3 @@
-ENV['BUNDLE_GEMFILE'] ||= File.expand_path('../../Gemfile', __FILE__)
+ENV['BUNDLE_GEMFILE'] ||= File.expand_path('../Gemfile', __dir__)
```
Comment (ninjapanzer) [inline config/boot.rb]: Missing magic comment `# frozen_string_literal: true`.

File: config/environments/production.rb [hunk 19]
```diff
@@ -14,33 +14,36 @@
...
+  # config.action_controller.asset_host = 'http://assets.example.com'
+
   # Specifies the header that your server uses for sending files.
   # config.action_dispatch.x_sendfile_header = 'X-Sendfile' # for Apache
   # config.action_dispatch.x_sendfile_header = 'X-Accel-Redirect' # for NGINX
 
+  # Mount Action Cable outside main process or domain
+  # config.action_cable.mount_path = nil
+  # config.action_cable.url = 'wss://example.com/cable'
+  # config.action_cable.allowed_request_origins = [ 'http://example.com', /http:\/\/example.*/ ]
```
Comment (ninjapanzer) [inline config/environments/production.rb]: Line is too long. [96/80]

File: config/environments/production.rb [hunk 20]
```diff
@@ -49,16 +52,15 @@
   config.log_level = :debug
 
   # Prepend all log lines with the following tags.
-  # config.log_tags = [ :subdomain, :uuid ]
-
-  # Use a different logger for distributed setups.
-  # config.logger = ActiveSupport::TaggedLogging.new(SyslogLogger.new)
+  config.log_tags = [ :request_id ]
```
Comment (ninjapanzer) [inline config/environments/production.rb]: Space inside square brackets detected.

File: config/environments/production.rb [hunk 21]
```diff
@@ -49,16 +52,15 @@
...
 
   # Use a different cache store in production.
   # config.cache_store = :mem_cache_store
 
-  # Enable serving of images, stylesheets, and JavaScripts from an asset server.
-  # config.action_controller.asset_host = 'http://assets.example.com'
+  # Use a real queuing backend for Active Job (and separate queues per environment)
```
Comment (ninjapanzer) [inline config/environments/production.rb]: Line is too long. [83/80]

File: config/environments/production.rb [hunk 22]
```diff
@@ -74,6 +76,16 @@
   # Use default logging formatter so that PID and timestamp are not suppressed.
   config.log_formatter = ::Logger::Formatter.new
 
+  # Use a different logger for distributed setups.
+  # require 'syslog/logger'
+  # config.logger = ActiveSupport::TaggedLogging.new(Syslog::Logger.new 'app-name')
```
Comment (ninjapanzer) [inline config/environments/production.rb]: Line is too long. [83/80]

File: config/environments/production.rb [hunk 23]
```diff
@@ -74,6 +76,16 @@
...
+
+  if ENV["RAILS_LOG_TO_STDOUT"].present?
```
Comment (ninjapanzer) [inline config/environments/production.rb]: Prefer single-quoted strings when you don't need string interpolation or special symbols.

File: config/environments/production.rb (hunk 22 above)
Comment (ninjapanzer) [inline config/environments/production.rb]: Line is too long. [83/80]

File: config/initializers/application_controller_renderer.rb [hunk 24]
```diff
@@ -0,0 +1,6 @@
+# Be sure to restart your server when you modify this file.
```
Comment (ninjapanzer) [inline config/initializers/application_controller_renderer.rb]: Missing magic comment `# frozen_string_literal: true`.

File: config/environments/production.rb (hunk 23 above)
Comment (ninjapanzer) [inline config/environments/production.rb]: Prefer single-quoted strings when you don't need string interpolation or special symbols.

File: config/initializers/application_controller_renderer.rb (hunk 24 above)
Comment (ninjapanzer) [inline config/initializers/application_controller_renderer.rb]: Missing magic comment `# frozen_string_literal: true`.

File: config/initializers/assets.rb [hunk 25]
```diff
@@ -3,13 +3,17 @@
...
 # Precompile additional assets.
-# application.js, application.css, and all non-JS/CSS in app/assets folder are already added.
-# Rails.application.config.assets.precompile += %w( search.js )
+# application.js, application.css, and all non-JS/CSS in the app/assets
+# folder are already added.
+# Rails.application.config.assets.precompile += %w( admin.js admin.css )
+
 # FRONT END ASSESTS
-Rails.application.config.assets.precompile += %w( posts.js posts_admin.js galleria.js reviews.js bricklayer_gallery.js )
+Rails.application.config.assets.precompile += %w( posts.js posts_admin.js galleria.js bricklayer_gallery.js )
```
Comment (ninjapanzer) [inline config/initializers/assets.rb]: `%w`-literals should be delimited by `[` and `]`.

Comment (ninjapanzer) [inline config/initializers/assets.rb]: `%w`-literals should be delimited by `[` and `]`.

Comment (ninjapanzer) [inline config/initializers/assets.rb]: Do not use spaces inside percent literal delimiters.

Comment (ninjapanzer) [inline config/initializers/assets.rb]: Do not use spaces inside percent literal delimiters.

Comment (ninjapanzer) [inline config/initializers/assets.rb]: Line is too long. [109/80]

Comment (ninjapanzer) [inline config/initializers/assets.rb]: Line is too long. [109/80]

File: config/initializers/backtrace_silencers.rb [hunk 26]
```diff
@@ -0,0 +1,7 @@
+# Be sure to restart your server when you modify this file.
```
Comment (ninjapanzer) [inline config/initializers/backtrace_silencers.rb]: Missing magic comment `# frozen_string_literal: true`.

Comment (ninjapanzer) [inline config/initializers/backtrace_silencers.rb]: Missing magic comment `# frozen_string_literal: true`.

File: config/initializers/backtrace_silencers.rb [hunk 27]
```diff
@@ -0,0 +1,7 @@
...
+
+# You can add backtrace silencers for libraries that you're using but don't wish to see in your backtraces.
```
Comment (ninjapanzer) [inline config/initializers/backtrace_silencers.rb]: Line is too long. [107/80]

Comment (ninjapanzer) [inline config/initializers/backtrace_silencers.rb]: Line is too long. [107/80]

File: config/initializers/backtrace_silencers.rb [hunk 28]
```diff
@@ -0,0 +1,7 @@
...
+# Rails.backtrace_cleaner.add_silencer { |line| line =~ /my_noisy_library/ }
+
+# You can also remove all the silencers if you're trying to debug a problem that might stem from framework code.
```
Comment (ninjapanzer) [inline config/initializers/backtrace_silencers.rb]: Line is too long. [112/80]

Comment (ninjapanzer) [inline config/initializers/backtrace_silencers.rb]: Line is too long. [112/80]

File: config/initializers/new_framework_defaults_5_1.rb [hunk 29]
```diff
@@ -0,0 +1,14 @@
+# Be sure to restart your server when you modify this file.
```
Comment (ninjapanzer) [inline config/initializers/new_framework_defaults_5_1.rb]: Missing magic comment `# frozen_string_literal: true`.

Comment (ninjapanzer) [inline config/initializers/new_framework_defaults_5_1.rb]: Missing magic comment `# frozen_string_literal: true`.

File: config/puma.rb [hunk 30]
```diff
@@ -0,0 +1,56 @@
+# Puma can serve each request in a thread from an internal thread pool.
```
Comment (ninjapanzer) [inline config/puma.rb]: Missing magic comment `# frozen_string_literal: true`.

Comment (ninjapanzer) [inline config/puma.rb]: Missing magic comment `# frozen_string_literal: true`.

File: config/puma.rb [hunk 31]
```diff
@@ -0,0 +1,56 @@
...
+# The `threads` method setting takes two numbers: a minimum and maximum.
+# Any libraries that use thread pools should be configured to match
+# the maximum value specified for Puma. Default is set to 5 threads for minimum
+# and maximum; this matches the default thread size of Active Record.
+#
+threads_count = ENV.fetch("RAILS_MAX_THREADS") { 5 }
```
Comment (ninjapanzer) [inline config/puma.rb]: Prefer single-quoted strings when you don't need string interpolation or special symbols.

Comment (ninjapanzer) [inline config/puma.rb]: Prefer single-quoted strings when you don't need string interpolation or special symbols.

File: Gemfile (hunk 2 above)
Comment (garyanewsome) [inline Gemfile]: Didn't know that

File: app/models/user.rb (hunk 11 above)
Comment (ninjapanzer) [inline app/models/user.rb]: NOOOOOOOO WHAT DID IT DO!!!

File: Gemfile (hunk 2 above)
Comment (ninjapanzer) [inline Gemfile]: I didn't either. Thanks PRONTO

File: app/models/user.rb (hunk 11 above)
Comment (ninjapanzer) [inline app/models/user.rb]: oh your type not mine. Yes in rails 5 we abstract ActiveRecord::Base as ApplicationRecord

File: bin/update (hunk 1 above)
Comment (ninjapanzer) [inline bin/update]: Similar code found in setup:14, update:14 (mass = 50)

File: app/controllers/reviews_controller.rb (hunk 3 above)
Comment (ninjapanzer) [inline app/controllers/reviews_controller.rb]: Space missing after colon.

Comment (ninjapanzer) [inline app/controllers/reviews_controller.rb]: Use `%i` or `%I` for an array of symbols.

File: app/controllers/sessions_controller.rb (hunk 4 above)
Comment (ninjapanzer) [inline app/controllers/sessions_controller.rb]: Prefer single-quoted strings when you don't need string interpolation or special symbols.

File: bin/yarn (hunk 14 above)
Comment (ninjapanzer) [inline bin/yarn]: Prefer single-quoted strings inside interpolations.

File: bin/yarn (hunk 15 above)
Comment (ninjapanzer) [inline bin/yarn]: Prefer single-quoted strings when you don't need string interpolation or special symbols.

File: bin/yarn (hunk 16 above)
Comment (ninjapanzer) [inline bin/yarn]: Prefer single-quoted strings when you don't need string interpolation or special symbols.

File: config/environments/production.rb (hunk 23 above)
Comment (ninjapanzer) [inline config/environments/production.rb]: Prefer single-quoted strings when you don't need string interpolation or special symbols.

File: config/initializers/assets.rb (hunk 25 above)
Comment (ninjapanzer) [inline config/initializers/assets.rb]: `%w`-literals should be delimited by `[` and `]`.

Comment (ninjapanzer) [inline config/initializers/assets.rb]: Do not use spaces inside percent literal delimiters.

File: config/puma.rb (hunk 31 above)
Comment (ninjapanzer) [inline config/puma.rb]: Prefer single-quoted strings when you don't need string interpolation or special symbols.

File: config/puma.rb [hunk 32]
```diff
@@ -0,0 +1,56 @@
...
+threads threads_count, threads_count
+
+# Specifies the `port` that Puma will listen on to receive requests; default is 3000.
+#
+port        ENV.fetch("PORT") { 3000 }
```
Comment (ninjapanzer) [inline config/puma.rb]: Prefer single-quoted strings when you don't need string interpolation or special symbols.

File: config/puma.rb [hunk 33]
```diff
@@ -0,0 +1,56 @@
...
+
+# Specifies the `environment` that Puma will run in.
+#
+environment ENV.fetch("RAILS_ENV") { "development" }
```
Comment (ninjapanzer) [inline config/puma.rb]: Prefer single-quoted strings when you don't need string interpolation or special symbols.

File: config/spring.rb [hunk 34]
```diff
@@ -0,0 +1,6 @@
+%w(
```
Comment (ninjapanzer) [inline config/spring.rb]: `%w`-literals should be delimited by `[` and `]`.

Comment (ninjapanzer) [inline config/spring.rb]: `%w`-literals should be delimited by `[` and `]`.

File: bin/update (hunk 1 above)
Comment (ninjapanzer) [inline bin/update]: Dont Care

File: app/controllers/reviews_controller.rb (hunk 3 above)
Comment (ninjapanzer) [inline app/controllers/reviews_controller.rb]: nope

File: bin/yarn (hunk 14 above)
Comment (ninjapanzer) [inline bin/yarn]: nope

File: bin/yarn (hunk 15 above)
Comment (ninjapanzer) [inline bin/yarn]: nope

File: bin/yarn (hunk 16 above)
Comment (ninjapanzer) [inline bin/yarn]: nope

File: config/initializers/assets.rb (hunk 25 above)
Comment (ninjapanzer) [inline config/initializers/assets.rb]: nope

File: config/initializers/assets.rb [hunk 35]
```diff
@@ -3,13 +3,17 @@
...
+# application.js, application.css, and all non-JS/CSS in the app/assets
+# folder are already added.
+# Rails.application.config.assets.precompile += %w( admin.js admin.css )
+
 # FRONT END ASSESTS
-Rails.application.config.assets.precompile += %w( posts.js posts_admin.js galleria.js reviews.js bricklayer_gallery.js )
+Rails.application.config.assets.precompile += %w(posts.js posts_admin.js galleria.js bricklayer_gallery.js)
 # BACKEND ASSESTS
-Rails.application.config.assets.precompile += %w( posts.css posts_admin.css )
+Rails.application.config.assets.precompile += %w(posts.css posts_admin.css)
```
Comment (ninjapanzer) [inline config/initializers/assets.rb]: `%w`-literals should be delimited by `[` and `]`.

**Pull request merged**